### DISCUSSION TITLE
[Feature] exploits database for nuclei

### DISCUSSION THREAD

**Discussion by ghost** [projectdiscovery/nuclei-templates#890] [General]
Hi guys,

Hope you are doing all well !

Just was wondering if [go-exploitdb](https://github.com/vulsio/go-exploitdb), a tool for searching Exploits from some Exploit Databases could be adapted for nuclei use case ?

Vulnerabilities database:
- ExploitDB(OffensiveSecurity) by CVE number or Exploit Database ID.
- GitHub Repositories
- Awesome Cve Poc

Sounds like a tremendous work to create all the scanning rules.

Thanks for any insights or inputs on that question.

Cheers,
Luc

**Comment by ehsandeep** [comment]
Hi @lucmichalski, 

Thank you for creating this issue, We are well aware of these well-known sources and have more to add in this lists, but unfortunately, we can not make use of them directly unless we covert them all into nuclei template format and that's going to be a **manual process**, for the time being, we are focusing on the developing the tool, we have been getting community contributions for exploits and many of them are from ExploitDB and few other projects, hope the list will grow with time. 

Let me know if you have any follow up questions/inputs.

**Comment by ghost** [comment]
Hi @bauthard ,

Thanks for your reply ! Greatly appreciated !

There is probably a way to pre-fill or at least assist the creation of these yaml files, like a wizard. 
I'll make some search on the topic, keep you updated.

Cheers,
Luc

**Comment by ehsandeep** [comment]
There is [something in making](https://twitter.com/pdnuclei/status/1303723108319199234) to ease the process of template creation 😄, unfortunately, we don't have an ETA to share.

**Comment by ghost** [comment]
Dude, it is counter-productive for your project/community as we could feed the nuclei's database :-) 
Please share !

Btw, have you ever considered to integrate Intel's hyperscan regex engine, works like a charm for me for >1k patterns.
Here is a golang implementation: https://github.com/flier/gohs

Cheers,
Luc
